src/main/java/org/prebid/server/bidder/adview/AdviewBidder.java

```diff
@@ -35,52 +40,93 @@
+    private final CurrencyConversionService currencyConversionService;
 
-    public AdviewBidder(String endpointUrl, JacksonMapper mapper) {
+    public AdviewBidder(String endpointUrl,
+                        JacksonMapper mapper,
+                        CurrencyConversionService currencyConversionService) {
         this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
         this.mapper = Objects.requireNonNull(mapper);
+        this.currencyConversionService = Objects.requireNonNull(currencyConversionService);
     }
 
     @Override
     public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
         final Imp firstImp = request.getImp().get(0);
         final ExtImpAdview extImpAdview;
+        final BidRequest modifiedBidRequest;
 
         try {
-            extImpAdview = mapper.mapper().convertValue(firstImp.getExt(), ADVIEW_EXT_TYPE_REFERENCE).getBidder();
-        } catch (IllegalArgumentException e) {
-            return Result.withError(BidderError.badInput("invalid imp.ext"));
+            extImpAdview = parseExtImp(firstImp);
+            modifiedBidRequest = modifyRequest(request, extImpAdview.getMasterTagId(), resolveBidFloor(request, firstImp));
+        } catch (PreBidException e) {
+            return Result.withError(BidderError.badInput(e.getMessage()));
         }
 
-        final BidRequest modifiedRequest = modifyRequest(request, extImpAdview.getMasterTagId());
         return Result.withValue(
                 HttpRequest.<BidRequest>builder()
                         .method(HttpMethod.POST)
                         .uri(resolveEndpoint(extImpAdview.getAccountId()))
                         .headers(HttpUtil.headers())
-                        .body(mapper.encodeToBytes(modifiedRequest))
-                        .payload(modifiedRequest)
+                        .body(mapper.encodeToBytes(modifiedBidRequest))
+                        .payload(modifiedBidRequest)
                         .build());
     }
 
-    private static BidRequest modifyRequest(BidRequest bidRequest, String masterTagId) {
+    private ExtImpAdview parseExtImp(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), ADVIEW_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException("invalid imp.ext");
+        }
+    }
+
+    private BigDecimal resolveBidFloor(BidRequest request, Imp imp) {
+        return shouldConvertBidFloor(imp.getBidfloor(), imp.getBidfloorcur())
+                ? convertBidFloorCurrency(imp.getBidfloor(), request, imp.getId(), imp.getBidfloorcur())
+                : imp.getBidfloor();
+    }
+
+    private static boolean shouldConvertBidFloor(BigDecimal bidFloor, String bidFloorCur) {
+        return BidderUtil.isValidPrice(bidFloor) && !StringUtils.equalsIgnoreCase(bidFloorCur, BIDDER_CURRENCY);
+    }
+
+    private BigDecimal convertBidFloorCurrency(BigDecimal bidFloor,
```

Comment:
i'd recommend to place `bidFloor` before `bidFloorCur`. We may not to adhere to `currencyConversionService` interface in client for better readability.